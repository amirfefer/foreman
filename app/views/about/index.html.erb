<% title _("About") %>

<div class="row" id="about">
  <div class="col-md-7">

    <div class="stats-well">
      <h4><%=_("System Status")%></h4>
      <div id="tabs"></div>   
      <%= mount_react_component('About', '#tabs', about_data.compact.to_json) %>
    </div>
  </div>
  <div class="col-md-5">
    <div class="stats-well" id="support">
      <h4><%= _("Support") %></h4>
      <h6><%= _("Documentation") %></h6>
      <ul>
        <li><%= link_to _('API Resources'), apipie_apipie_path, :title => _('Automate Foreman via a simple and powerful API') %></li>
        <li><%= link_to _('Manual'), documentation_url, :rel => "external" %></li>
        <li><%= link_to _('Wiki'), "http://projects.theforeman.org", :rel => "external" %></li>
      </ul>
      <h6><%= _("IRC") %></h6>
      <p><%= (_("You can find The Foreman on the %{freenode} (irc.freenode.net) network.  For general support, please visit #theforeman and for development specific related chat, please visit #theforeman-dev.") % {:freenode => link_to("freenode", "http://www.freenode.net", :rel => "external")}).html_safe  %></p>
      <h6><%= _("Mailing lists") %></h6>
      <p><%= _("Mailing lists are available via Google Groups. Much like IRC, we have a general users (support, Q/A, etc) lists and a development list:") %></p>
      <ul>
        <li><%= link_to _("Foreman Users"), "http://groups.google.com/group/foreman-users", :rel => "external" %></li>
        <li><%= link_to _("Foreman Developers"), "http://groups.google.com/group/foreman-dev", :rel => "external" %></li>
      </ul>
      <h6><%= _("Issue tracker") %></h6>
      <p><%= _("We use Redmine to report and track bugs and feature requests, which can be found here:") %> <%= link_to _("issue tracker"), "http://projects.theforeman.org/projects/foreman/issues", :rel => "external" %></p>
    </div>
    <div class="stats-well" id="copyright-div">
      <h4><%= _("System Information") %> </h4>
      <p id="copyright-p"><%= (_("Version %{version}  %{copyright} 2009-%{year} Paul Kelly and %{author}") % {:version => SETTINGS[:version], :copyright => "&copy;", :year=>DateTime.now.year, :author=>mail_to("ohadlevy@gmail.com", "Ohad Levy" )}).html_safe %></p>
    </div>
  </div>
</div>