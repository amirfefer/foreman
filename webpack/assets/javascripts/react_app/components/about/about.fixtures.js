export const data = {
  compute: [
    { id: { name: 'Libvirt', id: 1 }, type: 'Libvirt' },
    { id: { name: 'Ovirt', id: 2 }, type: 'Ovirt' },
  ],
  proxy: [
    {
      id: { name: 'regular', id: 1 },
      features: 'Facts, Logs, DNS, DHCP, and Puppet',
    },
  ],
  plugin: [
    {
      name: 'foreman_discovery',
      description: 'MaaS Discovery Plugin engine for Foreman',
      author:
        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ex ea difficultate illae fallaciloquae',
      version: '11.0.0',
    },
  ],
  provider: [
    { provider: 'Libvirt', status: true },
    { provider: 'oVirt', status: true },
    { provider: 'EC2', status: true },
    { provider: 'VMware', status: true },
    { provider: 'OpenStack', status: true },
    { provider: 'Rackspace', status: true },
    { provider: 'Google', status: true },
  ],
};
