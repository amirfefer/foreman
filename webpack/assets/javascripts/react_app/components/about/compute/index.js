import React from 'react';
import Table from '../../common/table';
import { columns } from './schema';
import { computeResource } from '../../../common/EmptyStates';

const AboutComputeTable = props => (
  <Table
    emptyState={computeResource()}
    rows={props.data}
    columns={columns} />
);

export default AboutComputeTable;
