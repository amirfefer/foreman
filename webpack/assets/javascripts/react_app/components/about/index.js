import React from 'react';
import TabsWrapper from '../common/tabs';
import AboutComputeTable from './compute';
import AboutPluginTable from './plugin';
import AboutProviderTable from './provider';
import AboutProxyTable from './proxies';

const About = ({ data }) => {
  const {
    compute, proxy, provider, plugin,
  } = data;
  const tabs = [
    __('Smart Proxies'),
    __('Available Providers'),
    __('Compute Resources'),
    __('Plugins'),
  ];

  return (
    <TabsWrapper id="about_tabs" tabs={tabs}>
      <AboutProxyTable data={proxy} />
      <AboutProviderTable data={provider} />
      <AboutComputeTable data={compute} />
      <AboutPluginTable data={plugin} />
    </TabsWrapper>
  );
};

export default About;
