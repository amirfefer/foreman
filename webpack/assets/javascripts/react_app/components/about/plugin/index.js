import React from 'react';
import Table from '../../common/table';
import { columns } from './schema';
import { plugin } from '../../../common/EmptyStates';

const AboutPluginTable = props => (
  <Table
    rows={props.data}
    columns={columns()}
    emptyState={plugin()}
  />
);

export default AboutPluginTable;
