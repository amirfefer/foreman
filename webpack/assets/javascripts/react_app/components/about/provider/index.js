import React from 'react';
import Table from '../../common/table';
import { columns } from './schema';
import { computeResource } from '../../../common/EmptyStates';

const AboutProviderTable = props => (
    <Table
      rows={props.data}
      emptyState={computeResource()}
      columns={columns()}
    />
);
export default AboutProviderTable;
