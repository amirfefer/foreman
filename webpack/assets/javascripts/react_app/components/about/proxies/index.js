import React from 'react';
import Table from '../../common/table';
import { columns } from './schema';
import { smartProxy } from '../../../common/EmptyStates';

const AboutProxyTable = props => (
  <Table
    emptyState={smartProxy()}
    rows={props.data}
    columns={columns()}
    />
);

export default AboutProxyTable;
