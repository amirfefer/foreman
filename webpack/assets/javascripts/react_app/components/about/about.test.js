import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import About from './';
import { data } from './about.fixtures';
import store from '../../redux';
import { mockRequest } from '../../mockRequests';

const flushAllPromises = () => new Promise(resolve => setImmediate(resolve));

describe('about page', () => {
  mockRequest({
    url: '/smart_proxies/1/ping',
    response: {
      success: true,
      message: {
        version: '1.16.0-develop',
        modules: {
          facts: '1.16.0',
          dns: '1.16.0',
          dhcp: '1.16.0',
          puppet: '1.16.0',
          logs: '1.16.0',
        },
      },
    },
  });
  mockRequest({
    url: '/compute_resources/1/ping',
    response: {
      status: 'OK',
      message: '',
    },
  });
  mockRequest({
    url: '/compute_resources/2/ping',
    response: {
      status: 'Error',
      message: 'Failed to open TCP connection',
    },
  });
  it('should render', () => {
    const wrapper = shallow(<About data={data} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('should have statuses and version fields', async () => {
    const wrapper = mount(<Provider store={store}>
        <About data={data} />
      </Provider>);

    await flushAllPromises();
    wrapper.update();
    expect(wrapper.find('#proxy1_version').text()).toBe('1.16.0-develop');
    expect(wrapper.find('#proxy1_status .pficon-ok')).toHaveLength(1);
    expect(wrapper.find('#compute_resource1_status .pficon-ok')).toHaveLength(1);
    expect(wrapper.find('#compute_resource2_status .pficon-error-circle-o')).toHaveLength(1);
  });
});
