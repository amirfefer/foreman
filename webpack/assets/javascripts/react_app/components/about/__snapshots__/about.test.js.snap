// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about page should render 1`] = `
<TabsWrapper
  id="about_tabs"
  tabs={
    Array [
      "Smart Proxies",
      "Available Providers",
      "Compute Resources",
      "Plugins",
    ]
  }
>
  <AboutProxyTable
    data={
      Array [
        Object {
          "features": "Facts, Logs, DNS, DHCP, and Puppet",
          "id": Object {
            "id": 1,
            "name": "regular",
          },
        },
      ]
    }
  />
  <AboutProviderTable
    data={
      Array [
        Object {
          "provider": "Libvirt",
          "status": true,
        },
        Object {
          "provider": "oVirt",
          "status": true,
        },
        Object {
          "provider": "EC2",
          "status": true,
        },
        Object {
          "provider": "VMware",
          "status": true,
        },
        Object {
          "provider": "OpenStack",
          "status": true,
        },
        Object {
          "provider": "Rackspace",
          "status": true,
        },
        Object {
          "provider": "Google",
          "status": true,
        },
      ]
    }
  />
  <AboutComputeTable
    data={
      Array [
        Object {
          "id": Object {
            "id": 1,
            "name": "Libvirt",
          },
          "type": "Libvirt",
        },
        Object {
          "id": Object {
            "id": 2,
            "name": "Ovirt",
          },
          "type": "Ovirt",
        },
      ]
    }
  />
  <AboutPluginTable
    data={
      Array [
        Object {
          "author": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Ex ea difficultate illae fallaciloquae",
          "description": "MaaS Discovery Plugin engine for Foreman",
          "name": "foreman_discovery",
          "version": "11.0.0",
        },
      ]
    }
  />
</TabsWrapper>
`;
